services/motion/builtin/utils.go

```diff
@@ -0,0 +1,532 @@
+	test.That(t, err, test.ShouldBeNil)
+
+	_, err = sens.DoCommand(ctx, map[string]interface{}{"use_compass": true})
+	test.That(t, err, test.ShouldBeNil)
+
+	_, err = sens.DoCommand(ctx, map[string]interface{}{"reset": true, "setLong": origin.Lng(), "setLat": origin.Lat()})
+	test.That(t, err, test.ShouldBeNil)
+	if noise != nil {
+		sens = &noisyMovementSensor{sens.(movementsensor.MovementSensor), noise, 0}
+	}
+
+	return sens.(movementsensor.MovementSensor)
+}
+
+func createMockSlamService(name, pcdPath string, origin spatialmath.Pose, move movementsensor.MovementSensor) (*inject.SLAMService, error) {
+	injectSlam := inject.NewSLAMService(name)
+	injectSlam.PointCloudMapFunc = func(ctx context.Context, returnEditedMap bool) (func() ([]byte, error), error) {
+		return getPointCloudMap(filepath.Clean(artifact.MustPath(pcdPath)))
+	}
+
+	gpsOrigin, _, err := move.Position(context.Background(), nil)
+	if err != nil {
+		return nil, err
+	}
+	if origin == nil {
+		origin = spatialmath.NewZeroPose()
+	}
+
+	injectSlam.PositionFunc = func(ctx context.Context) (spatialmath.Pose, error) {
+		currentGPS, _, err := move.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		headingLeft, err := move.CompassHeading(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		// Adjust for proper x-y control
+		headingLeft -= 90
+		// CompassHeading is a left-handed value. Convert to be right-handed. Use math.Mod to ensure that 0 reports 0 rather than 360.
+		heading := math.Mod(math.Abs(headingLeft-360), 360)
+		orientation := &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: heading}
+
+		geoPose := spatialmath.NewPose(spatialmath.GeoPointToPoint(currentGPS, gpsOrigin), orientation)
+
+		return spatialmath.Compose(origin, geoPose), nil
+	}
+	injectSlam.PropertiesFunc = func(ctx context.Context) (slam.Properties, error) {
+		return slam.Properties{
+			CloudSlam:             false,
+			MappingMode:           slam.MappingModeLocalizationOnly,
+			InternalStateFileType: ".pbstream",
+			SensorInfo: []slam.SensorInfo{
+				{Name: "my-camera", Type: slam.SensorTypeCamera},
+				{Name: "my-movement-sensor", Type: slam.SensorTypeMovementSensor},
+			},
+		}, nil
+	}
+	return injectSlam, nil
+}
+
+func createFrameSystemService(
+	ctx context.Context,
+	deps resource.Dependencies,
+	fsParts []*referenceframe.FrameSystemPart,
+	logger logging.Logger,
+) (framesystem.Service, error) {
+	fsSvc, err := framesystem.New(ctx, deps, logger)
+	if err != nil {
+		return nil, err
+	}
+	conf := resource.Config{
+		ConvertedAttributes: &framesystem.Config{Parts: fsParts},
+	}
+	if err := fsSvc.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	deps[fsSvc.Name()] = fsSvc
+
+	return fsSvc, nil
+}
+
+// CreateMoveOnGlobeEnvironment creates a testable environment that will simulate a real moving base for MoveOnGlobe calls.
+func CreateMoveOnGlobeEnvironment(ctx context.Context, t *testing.T, origin *geo.Point, noise spatialmath.Pose, sleepTime int) (
```

Comment:
Can we add the context of "Test" to the names of public functions in this file somehow? I think that would be useful for someone reading the docs for this package.  Also can we move public functions to the top of the file for increased readability?

Reply:
Yes.

They cannot be literally test functions, because then they will become private (even if capitalized, thanks Go) and they need to be accessed by the `motion-testing` repo, but I can certainly call them e.g. `CreateMoveOnGlobeTestEnvironment`